### PR TITLE
Create sitemap.xml

### PR DESCRIPTION
**Pull request by Userlastname**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @loqimean 
- [ ] @VasylenchukMischa

## Summary of issue

Create sitemap.xml

## Summary of change

Created sitemap.xml(uk/en)

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -24,6 +24,8 @@
     get "/calculator", to: "calculators#calculator"
     post "/receive_recomendations", to: "calculators#receive_recomendations"
     get "/about_us", to: redirect("/about_us.html")
+    get "/en/sitemap.xml", to: "sitemap#show", defaults: { format: "xml", locale: "en" }
```
Comment (mykhailovasylenchuk) [inline config/routes.rb:27]: А скоуп з локалями для чого?

File: spec/requests/sitemap.spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  describe "GET /show" do
+    it "renders the sitemap for en" do
+      get "/en/sitemap.xml"
+      expect(response).to be_successful
+    end
+
+    it "renders the sitemap for uk" do
+      get "/uk/sitemap.xml"
+      expect(response).to be_successful
+    end
```
Comment (loqimean) [inline spec/requests/sitemap.spec.rb:13]: прав

File: config/routes.rb (hunk 1 above)
Comment (loqimean) [reply config/routes.rb:27]: ага

**Review by loqimean — CHANGES_REQUESTED** [review]
коменти в чаті - прав

File: spec/requests/sitemap.spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require "rails_helper"
+
+RSpec.describe SitemapController, type: :request do
+  describe "GET /show" do
```
Comment (loqimean) [inline spec/requests/sitemap.spec.rb:4]:
```suggestion
  describe "GET #show" do
```

File: config/routes.rb [hunk 4]
```diff
@@ -24,6 +24,7 @@
     get "/calculator", to: "calculators#calculator"
     post "/receive_recomendations", to: "calculators#receive_recomendations"
     get "/about_us", to: redirect("/about_us.html")
+    get "/sitemap.xml", to: "sitemap#show", defaults: { format: "xml" }, as: :sitemap
```
Comment (loqimean) [inline config/routes.rb:27]: навіщо дефолт

File: app/controllers/sitemap_controller.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+class SitemapController < ApplicationController
+  # Method show render show.xml.builder for each locale
+  def show
+  end
```
Comment (loqimean) [inline app/controllers/sitemap_controller.rb:4]: де тести?

Comment (loqimean) [reply app/controllers/sitemap_controller.rb:4]: чому я бачу месидж, що нема тестів?

File: config/routes.rb [hunk 6]
```diff
@@ -20,7 +20,7 @@
                        skip: :omniauth_callbacks
 
     root "home#index"
-    get "/sitemap", to: "sitemap#index"
+    get "/sitemap", to: "sitemap#index", constraints: { format: /(html|xml)/ }
```
Comment (loqimean) [inline config/routes.rb:23]: навіщо ці констрейнти?

File: app/views/sitemap/index.xml.builder [hunk 7]
```diff
@@ -0,0 +1,20 @@
+xml.instruct! :xml, version: "1.0"
+xml.urlset "xmlns" => "http://www.sitemaps.org/schemas/sitemap/0.9", "xmlns:image" => "http://www.google.com/schemas/sitemap-image/1.1", "xmlns:video" => "http://www.google.com/schemas/sitemap-video/1.1" do
+  I18n.with_locale(locale) do
+    xml.url do
+      xml.loc root_url
+      xml.lastmod Time.now.iso8601
+      xml.changefreq "weekly"
```
Comment (loqimean) [inline app/views/sitemap/index.xml.builder:7]: ні

**Review by loqimean — APPROVED** [review]

**Pull request merged**